NEW: Check Thirdparty name

milenmk opened the pull request:
# New [*Check Thirdparty name*]

When adding a new client, check if a Third Party name already exists and throw an error if so.

milenmk committed:
Bug fix

When adding new client, check if Third Party name already exists and throw error if so.

stickler-ci committed:
Fixing style errors.

atm-maxime commented:
🤔 two thirdparties can have the same name...

milenmk committed:
fix check

milenmk committed:
Merge branch 'develop' of https://github.com/milenmk/dolibarr into develop

stickler-ci committed:
Fixing style errors.

eldy committed:
Update card.php

eldy commented on htdocs/societe/card.php:
```diff
@@ -406,6 +406,20 @@
 
+		if ($action == 'add') {
+			$thirdparty = GETPOST('name');
+			$sql = 'SELECT nom FROM '.MAIN_DB_PREFIX.'societe WHERE nom="'.$db->escape($thirdparty).'"';
```
It is frequent that 2 companies has the same name. 
Currenlty this is allowed into Dolibar but we may add an option to forget this.
You can add a test here 
if (! empty($conf->global->THIRDPARTY_NAME_MUST_BE_UNIQUE))

daraelmin replied:
What about individual ?  How would you manage namesake of humain being ?
How would you manage multiple business units, parent and child society with sans name ?

milenmk replied:
It is possible to add a filter to this, like 

`
if (! empty($conf->global->THIRDPARTY_NAME_MUST_BE_UNIQUE))
`

BB2A-Anthony commented:
It's interesing add option

And if existe proposed to create or not

It's interesing for product

milenmk committed:
vbvb

milenmk committed:
added global THIRDPARTY_NAME_MUST_BE_UNIQUE

If "THIRDPARTY_NAME_MUST_BE_UNIQUE" is set to 1, there cannot be 2 or more Thirdparties with same name. And id a Third Party name already exists - throw an error.

eldy commented:
This PR is now too old, so i close it. A lot of code may have been validated since the PR and the PR may be obsolete, or too difficult to merge due to other changes. Best choice is too cancel it to avoid to keep PR with too low chances to be validated.
